orm/table.go

```diff
@@ -95,9 +95,14 @@ type Table struct {
 func newTable(typ reflect.Type) *Table {
 	t := new(Table)
 	t.Type = typ
+	name := t.Type.Name()
+	index := strings.Index(name, "[")
+	if index > 0 {
+		name = name[:index]
+	}
 	t.zeroStruct = reflect.New(t.Type).Elem()
-	t.TypeName = internal.ToExported(t.Type.Name())
-	t.ModelName = internal.Underscore(t.Type.Name())
+	t.TypeName = internal.ToExported(name)
+	t.ModelName = internal.Underscore(name)
```

Comment:
Wouldn't an easier way to do this be to just define the name directly on the thing?

Like this:

```go
type File struct {
	tableName string `pg:"files"`

	...
```

Reply:
Directly defining the name works in some models without nested structure, but in some cases, it does not work.

example: https://go.dev/play/p/B3bcf3w_fxj

Reply:
Oh interesting, that is definitely a bug in my eyes then, the way table names are built should be consistent. Let me mess around a bit here. 

Thank you very much for the code reproduction!

Reply:
I see now what is happening, the way we generate the alias is different than how we derive the actual table name. Thus why it is only affecting select queries oddly.

I think there is a more elegant solution to this, but let me mess around some more.

Reply:
Have you found a better solution?